app/src/app/runner_schema/hybrid.rs

```diff
@@ -0,0 +1,204 @@
+        memory_cache_config: &MemoryCacheConfig,
+        repositories: Arc<HybridRepositoryModule>,
+    ) -> Self {
+        Self {
+            storage_config,
+            id_generator,
+            async_cache: memory::new_memory_cache(memory_cache_config),
+            memory_cache: MemoryCacheImpl::new(memory_cache_config, None),
+            repositories,
+        }
+    }
+}
+
+impl UseIdGenerator for HybridRunnerSchemaAppImpl {
+    fn id_generator(&self) -> &IdGeneratorWrapper {
+        &self.id_generator
+    }
+}
+impl UseRunnerSchemaParserWithCache for HybridRunnerSchemaAppImpl {
+    fn cache(&self) -> &MemoryCacheImpl<Arc<String>, RunnerSchemaWithDescriptor> {
+        &self.memory_cache
+    }
+}
+
+#[async_trait]
+impl RunnerSchemaApp for HybridRunnerSchemaAppImpl {
+    async fn create_runner_schema(
+        &self,
+        runner_schema: RunnerSchemaData,
+    ) -> Result<RunnerSchemaId> {
+        let id = RunnerSchemaId {
+            value: self.id_generator().generate_id()?,
+        };
+        let db = self.runner_schema_repository().db_pool();
+        let mut tx = db.begin().await.map_err(JobWorkerError::DBError)?;
+        let id = self
+            .runner_schema_repository()
+            .create(&mut *tx, id, &runner_schema)
+            .await?;
+        tx.commit().await.map_err(JobWorkerError::DBError)?;
+        let _ = self
+            .delete_cache_locked(&Self::find_all_list_cache_key())
+            .await;
+
+        Ok(id)
+    }
+
+    // offset, limit 付きのリストキャッシュは更新されないので反映に時間かかる (ttl短かめにする)
+    async fn update_runner_schema(
+        &self,
+        id: &RunnerSchemaId,
+        runner_schema: &Option<RunnerSchemaData>,
+    ) -> Result<bool> {
+        if let Some(w) = runner_schema {
+            let pool = self.runner_schema_repository().db_pool();
+            let mut tx = pool.begin().await.map_err(JobWorkerError::DBError)?;
+            self.runner_schema_repository()
+                .update(&mut *tx, id, w)
+                .await?;
+            tx.commit().await.map_err(JobWorkerError::DBError)?;
+            // clear memory cache
+            let _ = self
+                .delete_cache_locked(&Self::find_cache_key(&id.value))
+                .await;
+            let _ = self
+                .delete_cache_locked(&Self::find_all_list_cache_key())
+                .await;
+            Ok(true)
+        } else {
+            // all empty, no update
+            Ok(false)
+        }
+    }
+
+    async fn delete_runner_schema(&self, id: &RunnerSchemaId) -> Result<bool> {
+        let res = self.runner_schema_repository().delete(id).await?;
+        // clear memory cache
+        let _ = self
+            .delete_cache_locked(&Self::find_cache_key(&id.value))
+            .await;
+        let _ = self
+            .delete_cache_locked(&Self::find_all_list_cache_key())
+            .await;
+        Ok(res)
+    }
+
+    async fn find_runner_schema(
```

Comment:
Returns protobufs as information to create forms for display and editing on the web front.